File: src/test/java/com/uapp_llc/test/stub/repository/ColumnRepositoryStub.java
package com.uapp_llc.test.stub.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageImpl;
import org.springframework.data.domain.Pageable;

import com.uapp_llc.model.Column;
import com.uapp_llc.repository.ColumnRepository;
import com.uapp_llc.test.stub.repository.identification.Identification;

public class ColumnRepositoryStub
    extends IdentifyingRepositoryStub<Long, Column>
    implements ColumnRepository {

  public ColumnRepositoryStub(Identification<Column> identification) {
    super(identification);
  }

  @Override
  protected Long getId(Column entity) {
    return entity.getId();
  }

  @Override
  public Column save(Column entity) {
    return super.save(entity);
  }

  @Override
  public Optional<Column> findById(Long id) {
    Column entity = super.find(id);
    return Optional.ofNullable(entity);
  }

  @Override
  public List<Column> findAll() {
    return super.findAll();
  }

  @Override
  public Page<Column> findAll(Pageable pageable) {
    List<Column> entities = super.findAll();
    return new PageImpl<>(entities);
  }

  @Override
  public int count() {
    return super.size();
  }

  @Override
  public void delete(Column entity) {
    super.delete(entity);
  }

}

File: src/test/kotlin/com/uapp_llc/test/stub/repository/ColumnRepositoryStub.kt
package com.uapp_llc.test.stub.repository;

import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import com.uapp_llc.model.Column
import com.uapp_llc.repository.ColumnRepository
import com.uapp_llc.test.stub.repository.identification.Identification

class ColumnRepositoryStub(

    identification: Identification<Column>

) : IdentifyingRepositoryStub<Long, Column>(identification),
    ColumnRepository {

  override fun getId(entity: Column): Long? = entity.id

  override fun save(entity: Column): Column = super.save(entity)

  override fun findById(id: Long): Column? = super.find(id)

  override fun findAll(pageable: Pageable): Page<Column> = PageImpl(super.findAll())

  override fun findAllByOrderByIndex(): MutableList<Column> =
      super.findAll()
          .sortedBy { it.index }
          .toMutableList()

  override fun count(): Int = super.size()

  override fun delete(entity: Column) = super.delete(entity)

}
